### PR TITLE
MCP23017 support.

### PR DESCRIPTION
**Pull request by lobradov**
## Description:

MCP23017 support.

**Note: I'm not very good with Python. Please double-check before merging.**

**Related issue:** implements OttoWinter/esphomelib#162

**Pull request in [esphomedocs](https://github.com/OttoWinter/esphomedocs) with documentation (if applicable):** OttoWinter/esphomedocs#<esphomedocs PR number goes here>
**Pull request in [esphomelib](https://github.com/OttoWinter/esphomelib) with C++ framework changes (if applicable):** OttoWinter/esphomelib#218

## Example entry for YAML configuration (if applicable):
```yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [X] Tests have been added to verify that the new code works (under `tests/` folder).
  - [X] Check this box if you have read, understand, comply, and agree with the [Code of Conduct](https://github.com/OttoWinter/esphomeyaml/blob/master/CODE_OF_CONDUCT.md).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphomedocs](https://github.com/OttoWinter/esphomedocs).

### CODE REVIEW

File: esphomeyaml/helpers.py [hunk 1]
```diff
@@ -585,15 +585,21 @@ def generic_gpio_pin_expression_(conf, mock_obj, default_mode):
...
             mode = conf.get(CONF_MODE, u'INPUT')
-            yield hub.make_input_pin(number,
+            if mux == CONF_PCF8574:
+                yield hub.make_input_pin(number,
                                      RawExpression('PCF8574_' + mode),
                                      inverted)
+            elif mux == CONF_MCP23017:
+                yield hub.make_input_pin(number,
+                                      RawExpression('MCP23017_' + mode),
+                                      inverted)
```
Comment (OttoWinter) [inline esphomeyaml/helpers.py]: Indent this with 3 more spaces to fix the lint error:

```python
yield hub.make_input_pin(number,
                         RawExpression('MCP23017_' + mode),
                         inverted)
```

**Review by OttoWinter — COMMENTED** [review]
Looks good!

File: esphomeyaml/helpers.py (hunk 1 above)
Comment (lobradov) [reply esphomeyaml/helpers.py]: ... there's a good reason why I never liked COBOL or Python... :)

**Comment by OttoWinter** [comment]
Closing due to inactivity.

**Pull request closed without merging**